lib/ace/transport_app.rb

```diff
@@ -93,28 +96,33 @@ def validate_schema(schema, body)
       error = validate_schema(@schemas["execute_catalog"], body)
       return [400, error.to_json] unless error.nil?
 
-      # simulate expected error cases
-      if body['compiler']['certname'] == 'fail.example.net'
-        [200, { _error: {
-          msg: 'catalog compile failed',
-          kind: 'puppetlabs/ace/compile_failed',
-          details: 'upstream api errors go here'
-        } }.to_json]
-      elsif body['compiler']['certname'] == 'credentials.example.net'
-        [200, { _error: {
-          msg: 'target specification invalid',
-          kind: 'puppetlabs/ace/target_spec',
-          details: 'upstream api errors go here'
-        } }.to_json]
-      elsif body['compiler']['certname'] == 'reports.example.net'
-        [200, { _error: {
-          msg: 'report submission failed',
-          kind: 'puppetlabs/ace/reporting_failed',
-          details: 'upstream api errors go here'
-        } }.to_json]
-      else
-        [200, '{}']
-      end
+      # TODO: retrieve these from the target
+      facts = {}
+      trusted_facts = {}
+
+      http_response = @client.retrieve(
+        body['compiler']['certname'],
+        body['compiler']['environment'],
+        facts,
+        trusted_facts,
+        body['compiler']['transaction_uuid'],
+        body['compiler']['job_id']
+      )
+
+      response = if http_response.is_a? ACE::Error
+                   {
+                     status: 200,
+                     json: { _error: http_response }
+                   }
+                 else
+                   {
+                     status: http_response.code,
```

Comment:
Will we not always be returning a 200, since calling the endpoint was still be a success at this point?